src/live_restore/live_restore_fs.c

```diff
@@ -589,7 +592,24 @@ __live_restore_fs_fill_holes_on_file_close(WT_FILE_HANDLE *fh, WT_SESSION *wt_se
         __wt_verbose_debug3((WT_SESSION_IMPL *)wt_session, WT_VERB_FILEOPS,
           "Found hole in %s at %ld-%ld during file close. Filling", fh->name, hole->off,
           WT_EXTENT_END(hole));
-        WT_RET(__live_restore_fh_read(fh, wt_session, hole->off, hole->len, buf));
+
+        off_start = hole->off;
+        off_end = WT_EXTENT_END(hole);
+
+        /*
+         * When encountering a large hole, break the read into small chunks. Split the hole into
+         * n chunks: the first n - 1 chunks will read a full WT_LIVE_RESTORE_READ_SIZE buffer, and
+         * the last chunk reads the remaining data.
+         */
+        while (off_start + WT_LIVE_RESTORE_READ_SIZE <= off_end) {
+            WT_RET(
+              __live_restore_fh_read(fh, wt_session, off_start, WT_LIVE_RESTORE_READ_SIZE, buf));
+            off_start += (wt_off_t)WT_LIVE_RESTORE_READ_SIZE;
+        }
+        /* Read the last chunk */
+        WT_RET(__live_restore_fh_read(
```

Comment:
If the previous loop exists because `off_start + WT_LIVE_RESTORE_READ_SIZE = off_end` then you'll do a size 1 read here. That seems wrong.

Reply:
I don't think this is wrong, a range [l, r] will have a length of r - l + 1. If `off_start + WT_LIVE_RESTORE_READ_SIZE = off_end` that means we are having `WT_LIVE_RESTORE_READ_SIZE + 1` size not read yet, thus an extra size 1 read is necessary. 
E.g. If we have
```
off_start = 10;
WT_LIVE_RESTORE_READ_SIZE = 10;
off_end = 20;
```
In the above case `off_start + WT_LIVE_RESTORE_READ_SIZE == off_end`, however we need to call `__live_restore_fh_read` twice, the first for offset [10,19] and the second for offset [20,20].